From: Loic Dachary <ldachary@redhat.com>
Date: Tue, 12 Jul 2016 16:56:52 +0200
Subject: ceph-disk: timeout ceph-disk to avoid blocking forever

When ceph-disk runs from udev or init script, it is in the background
and should it block for any reason, it may keep a lock forever. All
calls to ceph-disk in these context are changed to timeout.

The TimeoutStartSec= and TimeoutStopSec= which are both set via
TimeoutSec= do not apply to Type=oneshot services.

https://www.freedesktop.org/software/systemd/man/systemd.service.html

Fixes: http://tracker.ceph.com/issues/16580

Signed-off-by: Loic Dachary <loic@dachary.org>
(cherry picked from commit bed1a5cc05a9880b91fc9ac8d8a959efe3b3d512)
(cherry picked from commit 430ab1b83e67dfb697b034e669b06b7a600bcc6b)

Resolves: rhbz#1351484
---
 systemd/ceph-disk@.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/systemd/ceph-disk@.service b/systemd/ceph-disk@.service
index 8b18ba4..f13c30b 100644
--- a/systemd/ceph-disk@.service
+++ b/systemd/ceph-disk@.service
@@ -4,5 +4,5 @@ Description=Ceph disk activation: %f
 [Service]
 Type=oneshot
 KillMode=none
-ExecStart=/bin/sh -c 'flock /var/lock/ceph-disk /usr/sbin/ceph-disk --verbose --log-stdout trigger --sync %f'
+ExecStart=/bin/sh -c 'timeout 120 flock /var/lock/ceph-disk /usr/sbin/ceph-disk --verbose --log-stdout trigger --sync %f'
 TimeoutSec=0
